### PR TITLE
create card-grid block

### LINKED ISSUE #36: Card block: support large card on the right
See https://www.24life.com/magazine/volume-5-issue-11/

Extension of https://github.com/hlxsites/24life/issues/3

Suggestions: 
- Check the order of articles, and if the large article is the first, put it on the left. Otherwise put it on the right. 
- or automatically alternative between left and right after each group of blocks. 
- alternatively, let the author add a 'large right' variant to the block. 

Screenshot 2023-08-01 at 12 11 19

### PR DESCRIPTION
**Pull request by rrusher**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #36 

Test URLs:
- Before: https://main--24life--hlxsites.hlx.page/drafts/rrusher/large-card
- After: https://3-card-large--24life--hlxsites.hlx.page/drafts/rrusher/large-card

### CODE REVIEW

File: blocks/card-grid/card-grid.css [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  height: 100%;
+  object-fit: cover;
+}
+
+.block.card-grid>ul>li>div {
+  position: relative;
+}
+
+.block.card-grid>ul>li h3 {
+  font-family: var(--ff-volvo-novum-medium);
```
Comment (synox) [inline blocks/card-grid/card-grid.css:73]: ehm.. volvo?

File: blocks/card-grid/card-grid.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          }
+        });
+        cell.remove();
+      }
+    });
+    if (rowIndex > 0) {
+      row.remove();
+    }
+  });
+  removeEmptyLi(block);
```
Comment (synox) [inline blocks/card-grid/card-grid.js:37]: Please don't put lists in table cells. Instead use on cell for each link. Then you can also remove the ugly <li> handling in this code.

File: blocks/card-grid/card-grid.css [hunk 3]
```diff
@@ -0,0 +1,97 @@
+.section .card-grid-wrapper {
```
Comment (synox) [inline blocks/card-grid/card-grid.css:1]: We already have the layout implemented in card, see https://main--24life--hlxsites.hlx.page/tools/sidekick/library.html?plugin=blocks&path=/block-library/blocks/card&index=2
It does not yet do large on the right side yet. 

We could reuse that feature here. Then this block would populate the cards from the links, and control the layout, but could be simpler? WDYT?

File: blocks/card-grid/card-grid.js
Comment (synox) [inline blocks/card-grid/card-grid.js:1]: It's flaky, on every few reloads the order is different, and sometimes the whole layout is broken. 

https://github.com/hlxsites/24life/assets/2250964/1577dc24-bd14-4ede-85ce-7e7b280b9c1d

**Review by synox — CHANGES_REQUESTED** [review]

File: blocks/card-grid/card-grid.css (hunk 1 above)
Comment (rrusher) [reply blocks/card-grid/card-grid.css:73]: borrowed code, corrected.

File: blocks/card-grid/card-grid.css (hunk 3 above)
Comment (rrusher) [reply blocks/card-grid/card-grid.css:1]: Yes. I didn't realize it was there.

File: blocks/card-grid/card-grid.js
Comment (rrusher) [reply blocks/card-grid/card-grid.js:1]: Fixed.

File: blocks/card-grid/card-grid.js (hunk 2 above)
Comment (rrusher) [reply blocks/card-grid/card-grid.js]: removed everything and rebuild with minimal nodes.

Comment (rrusher) [reply blocks/card-grid/card-grid.js]: fixed

**Comment by rrusher** [comment]
Pulling everything for a card from just the url was not performing well enough to keep it.
With this many articles, it was just too slow.
Abandoning this block.

**Comment by synox** [comment]
@rrusher maybe the magazine-issue page can fetch all the articles for that issue at once. having a filter option in spreadsheets would be very useful for this

**Pull request closed without merging**